js/toDoController.js

```diff
@@ -0,0 +1,25 @@
+ToDoList.controller('ToDoController', ['$scope', function($scope) {
+
+  $scope.todos = [];
+
+  $scope.submitTodo = function(){
+    $scope.todos.push({'title':$scope.newTodo, 'complete':false});
+    $scope.newTodo = '';
+  };
+  $scope.clearCompletedTasks = function(){
+    $scope.todos = $scope.todos.filter(function(task){
+      return !task.complete;
```

Comment:
`$scope.todos = ` is actually rewriting the view model every time you clear completed tasks - so the change is permanent.